logstash-rs/src/buffer.rs

```diff
@@ -39,18 +41,29 @@ impl BufferedSender {
 
 impl Sender for BufferedSender {
     fn send(&self, event: LogStashRecord) -> Result<()> {
-        self.sender.send(Command::Send(event)).map_err(From::from)
+        let result = self.sender.try_send(Command::Send(event));
+        process_result(result, event.level <= Level::Warn)
     }
 
     fn send_batch(&self, events: Vec<LogStashRecord>) -> Result<()> {
-        self.sender.send(Command::SendBatch(events))?;
-        Ok(())
+        let important = events.iter().any(|e| e.level <= Level::Warn);
+        let result = self.sender.try_send(Command::SendBatch(events));
+        process_result(result, important)
     }
 
     fn flush(&self) -> Result<()> {
-        self.sender.send(Command::Flush)?;
-        Ok(())
+        let result = self.sender.try_send(Command::Flush);
+        process_result(result, false)
+    }
+}
+
+fn process_result<T>(r: std::result::Result<(), TrySendError<T>>, log_full: bool) -> Result<()> {
+    if matches!(r, Err(TrySendError::Disconnected(..))) {
+        return Err(Error::SenderThreadStopped(r.unwrap_err().to_string()));
+    } else if (log_full && matches!(r, Err(TrySendError::Full(..)))) {
```

Comment:
I think, it is better to do a single match, that checks for both `Disconnected` and `Full`